tests/vertexonly/test_interpolation_from_parent.py

```diff
@@ -336,31 +342,31 @@ def test_mixed_function_interpolation(parentmesh, vertexcoords, tfs):
     # Enough tests - don't both using it again for a different Function in V
 
 
-def test_scalar_real_interpolation(parentmesh, vertexcoords):
+def test_scalar_real_interpolation(parentmesh, vertexcoords, constant_function):
     vm = VertexOnlyMesh(parentmesh, vertexcoords, missing_points_behaviour=None)
     W = FunctionSpace(vm, "DG", 0)
     V = FunctionSpace(parentmesh, "Real", 0)
     # Remove below when interpolating constant onto Real works for extruded
     if type(parentmesh.topology) is mesh.ExtrudedMeshTopology:
         with pytest.raises(ValueError):
-            interpolate(Constant(1.0, domain=parentmesh), V)
```

Comment:
Does this not work if you just remove the `domain=parentmesh` bit? I don't strictly see why we need `constant_function` (a real `Function`) instead of a `Constant` here.

Reply:
I think the domain is important for the adjoints (but we are not testing that here)